### PR TITLE
[SYCL-MLIR]: Allow cgeist to pass -mllvm to BE toolchain.

### PR DESCRIPTION
**Pull request by etiotto**
This PR modifies the `cgeist` driver to register the `-mllvm <value>` option and forward it to the LLVM toolchain.
This is an essential ability required to trace/debug LLVM transformations performed by the LLVM backend toolchain on the LLVM IR `cgeist` generates.

### CODE REVIEW

File: polygeist/tools/cgeist/driver.cc [hunk 1]
```diff
@@ -1040,70 +1062,94 @@ int main(int argc, char **argv) {
...
 
-  int size = MLIRArgs.size();
-  const char **data = MLIRArgs.data();
-  InitLLVM y(size, data);
-  llvm::cl::ParseCommandLineOptions(size, data);
+  // Register command line options specific to cgeist.
+  {
+    int Size = MLIRArgs.size();
+    const char **Data = MLIRArgs.data();
+    InitLLVM Y(Size, Data);
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/driver.cc:1110]: Why are you putting the `InitLLVM` object in a scope? 
We probably need it to be in the same scope as `main()` for `-debug-only=` to work.

File: polygeist/tools/cgeist/driver.cc [hunk 2]
```diff
@@ -718,7 +720,29 @@ runOptimizationPipeline(llvm::Module &module,
           ? PB.buildO0DefaultPipeline(OptimizationLevel)
           : PB.buildPerModuleDefaultPipeline(OptimizationLevel);
 
-  MPM.run(module, MAM);
+  // Before executing passes, print the final values of the LLVM options.
+  cl::PrintOptionValues();
+
+  // Print a textual representation of pipeline ther LLVM pipeline.
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/driver.cc:726]: typo: ther

**Comment by victor-eds** [comment]
Can we have a test for this functionality?

File: polygeist/tools/cgeist/driver.cc (hunk 1 above)
Comment (etiotto) [reply polygeist/tools/cgeist/driver.cc]: done

File: polygeist/tools/cgeist/driver.cc [hunk 3]
```diff
@@ -718,7 +720,29 @@ runOptimizationPipeline(llvm::Module &module,
           ? PB.buildO0DefaultPipeline(OptimizationLevel)
           : PB.buildPerModuleDefaultPipeline(OptimizationLevel);
 
-  MPM.run(module, MAM);
+  // Before executing passes, print the final values of the LLVM options.
+  cl::PrintOptionValues();
+
+  // Print a textual representation of pipeline the LLVM pipeline.
```
[the remark this answers is not in the document]
Comment (etiotto) [reply polygeist/tools/cgeist/driver.cc]: done

**Review by whitneywhtsang — APPROVED** [review]

**Comment by etiotto** [comment]
@victor-eds does this looks OK to you. I have added a lit test.

**Review by victor-eds — APPROVED** [review]

**Pull request merged**